mimalloc-sys/build.rs

```diff
@@ -1,64 +1,50 @@
-//! Builds and links mimalloc
+//! Builds and links mimalloc.  Emulates behavior of CMakeLists
+// Based on flate2-rs's miniz-sys
 
-macro_rules! info {
-    ($($args:tt)*) => { println!($($args)*) }
-}
-
-use std::{env, fs, path::PathBuf};
+use std::env;
 
 fn main() {
     // Get environment variables:
     let target = env::var("TARGET").expect("TARGET was not set");
-    let host = env::var("HOST").expect("HOST was not set");
-    let profile = env::var("PROFILE").expect("PROFILE was not set");
-    let num_jobs = env::var("NUM_JOBS").expect("NUM_JOBS was not set");
-    let out_dir =
-        PathBuf::from(env::var_os("OUT_DIR").expect("OUT_DIR was not set"));
-    let src_dir = env::current_dir().expect("failed to get current directory");
-    let build_dir = out_dir.join("build");
-    let mimalloc_src_dir = src_dir.join("mimalloc");
-    let mimalloc_out_src_dir = build_dir.join("mimalloc");
 
-    info!("TARGET={}", target.clone());
-    info!("HOST={}", host.clone());
-    info!("PROFILE={}", profile.clone());
-    info!("NUM_JOBS={}", num_jobs.clone());
-    info!("OUT_DIR={:?}", out_dir);
-    info!("BUILD_DIR={:?}", build_dir);
-    info!("SRC_DIR={:?}", src_dir);
-    info!("mimalloc_src_dir={:?}", mimalloc_src_dir);
-    info!("mimalloc_out_src_dir={:?}", mimalloc_out_src_dir);
+    // Build mimalloc
+    let mut build = cc::Build::new();
+
+    build
+        .file("mimalloc/src/static.c")
+        .include("mimalloc/include")
+        .include("mimalloc/src") // For .c includes from static.c and page.c
+        .define("MI_STATIC_LIB", "1") // Tell mimalloc this is a static build
+        .warnings(false);
+
+    if !target.contains("darwin") && !target.contains("windows") {
+        build.flag("-fvisibility=hidden");
+    }
+
```

Comment:
When statically linking the library, if the compiler is clang then `-C cross-lang-lto` should be used to be able to inline the library into Rust.

Reply:
I couldn't find documentation on this.  Is it a rustc flag?  clang didn't like it.

Reply:
https://github.com/rust-lang/rust/blob/master/src/test/run-make-fulldeps/cross-lang-lto-clang/Makefile#L23 The option was renamed to `-C linker-plugin-lto=on`.